upgrades/schema/Version_6_0_20210908142400_add_product_web_api_acl.php

```diff
@@ -0,0 +1,305 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Pim\Upgrade\Schema;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+use Oro\Bundle\SecurityBundle\Acl\Persistence\AclManager;
+use Symfony\Component\DependencyInjection\ContainerAwareInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * @copyright 2021 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+final class Version_6_0_20210908142400_add_product_web_api_acl extends AbstractMigration implements ContainerAwareInterface
+{
+    private const ACLS = [
+        'pim_api_product_list' => true,
+        'pim_api_product_edit' => true,
+        'pim_api_product_remove' => true,
+    ];
+
+    private ?ContainerInterface $container;
+
+    public function up(Schema $schema): void
+    {
+        $this->disableMigrationWarning();
+
+        $roles = $this->getRoles();
+
+        foreach (self::ACLS as $acl => $granted) {
+            if (!$this->aclIsRegistered($acl)) {
+                $this->registerAcl($acl);
+            }
+
+            foreach ($roles as $role) {
+                if ($this->aclIsAlreadyDefinedForRole($role, $acl)) {
+                    continue;
+                }
+
+                if ($granted && $this->roleIsGrantedEverythingByDefault($role)) {
+                    continue;
+                }
+
+                $this->addAclToRole($role, $acl, $granted);
+            }
+        }
+
+        /** @var AclManager $aclManager */
+        $aclManager = $this->container->get('oro_security.acl.manager');
+        $aclManager->clearCache();
+    }
+
+    private function aclIsRegistered(string $acl): bool
+    {
+        return (bool) $this->connection->fetchOne(
+            <<<SQL
+SELECT COUNT(*)
+FROM acl_classes
+WHERE class_type = :acl
+SQL,
+            [
+                'acl' => $acl,
+            ]
+        );
+    }
+
+    private function registerAcl(string $acl): void
+    {
+        $this->connection->executeQuery(
+            <<<SQL
+INSERT INTO acl_classes (class_type)
+VALUES (:acl)
+SQL,
```

Comment:
```suggestion
            'INSERT INTO acl_classes (class_type) VALUES (:acl)',
```
Just to say we can also use simple strings...

Reply:
As I said, I prefer to write the SQL queries like this.
This is what we do usually in the PIM, since a long time.

If we decide to add a different rule about it, I will comply without issue.
For the current PR, I don't feel like rewriting it.

Also, regarding the fact that there is a lot of queries in this migration and it can be visually annoying, I don't want to hide the fact that the current tables for the ACLs are much harded to migrate than necessary.

I hope you don't mind that I resolve your comments about CS.